pkg/apis/core/types.go

```diff
@@ -335,6 +335,14 @@ type PersistentVolumeSpec struct {
 	// This field influences the scheduling of pods that use this volume.
 	// +optional
 	NodeAffinity *VolumeNodeAffinity
+	// Name of VolumeAttributesClass to which this persistent volume belongs. Empty value
+	// is not allowed. Use nil to indicate that this volume does not belong to any
```

Comment:
What happens if the VolumeAttributesClass doesn't exist? Can we document that? I'd expect this to happen both when the class name is mistyped and when the VolumeAttributesClass is created after this field is set.  In the first case, it would be really helpful to provide user feedback. In the second case, it would be good to be as graceful and not transiently break anything (e.g. when a "kubectl apply" changes multiple files, but the order of updates results in a broken reference for a short period of time)

Reply:
With the new status updates proposed in https://github.com/kubernetes/enhancements/pull/4298, the status will be Pending and a condition will have the more detailed message. @carlory can you update the comments here?

Reply:
For this one, we added to the status field to explain what pending means:
https://github.com/kubernetes/kubernetes/pull/121104/files#diff-19b27e79a2dcb337f58b0f7f3477a29d29498b5d135fbb67e9f858cf1dae222cR577

Do we want to restate it here?

Reply:
@carlory Let's add one more sentence to explain what happens if the VolumeAttributesClass does not exist.

Reply:
@sunnylovestiramisu @msau42 Sorry, I don't understand what shoud I do. Do you mean to update the volumeAttributesClassName field's description in the **PV** definition or something else?

Reply:
We should update in staging/src/[k8s.io/api/core/v1/types.go](http://k8s.io/api/core/v1/types.go) and pkg/apis/core/types.go and they will be in sync.

Reply:
I've added some suggestion comments to this PR that I believe should resolve this.